assets/javascripts/discourse/templates/components/gamification-leaderboard-row.hbs

```diff
@@ -9,7 +9,13 @@
     data-user-card={{rank.username}}
   >
     {{avatar rank imageSize="large"}}
-    <span class="user__name">{{rank.username}}</span>
+    <span class="user__name">
+      {{#if this.siteSettings.prioritize_username_in_ux}}
+        {{rank.username}}
+      {{else}}
+        {{rank.name}}
+      {{/if}}
+    </span>
```

Comment:
Can we have a test for this logic to protect from regressions?

Reply:
A5FABF42-2F80-4F7E-BD28-F034D1023836

Reply:
I just added some tests. Please let me know if there are any issues!